### ISSUE TITLE
In a Vue front-end project, when trying to use Playwright-crx, it fails to work and the API calls are not successful.

### ISSUE THREAD

**Issue by Xiaomajohn** [ruifigueira/playwright-crx#58]
How can I do for it?
Do we need to import source code offline?

**Comment by ruifigueira** [comment]
Playwright-crx only runs inside an extension service worker. If you try to use it directly in the frontend it won't work, unfortunately. You can nevertheless use extension [messaging](https://developer.chrome.com/docs/extensions/develop/concepts/messaging) for the extension to communicate with frontend.

**Comment by Xiaomajohn** [comment]
However, in the background program of the extension plugin, I found that in 
`playwright-crx`, the `const crxApp = await crx.start({ slowMo: 500 })` 
this line could not obtain the specified time after packaging, and it prompted an error.
![image](https://github.com/user-attachments/assets/483da768-e566-4762-8f91-266a57f4fe88)
import { crx } from 'playwright-crx';
import { createTodos } from './todos.ts';

chrome.runtime.onMessage.addListener(async (request) => {
  if (request.type === 'debug_url') {
    chrome.runtime.sendMessage({ type: 'debug_url', url: request.tab?.url });
    await chrome.action.disable();
    console.log('first');
    const crxApp = await crx.start({ slowMo: 500 });
    console.log('second');
    const page = await crxApp.attach(request.tab?.id!).catch(() => crxApp.newPage());
    console.log('3');

    try {
      await createTodos(page);
    } finally {
      await crxApp.close();
      await chrome.action.enable();
    }
  }
});
chrome.runtime.onMessage.addListener(async (request) => {
  if (request.type === 'COUNT') {
    console.log('COUNT', request.count);
  }
});

**Comment by ruifigueira** [comment]
Are you using vite to bundle the extension? Does the error provide more information?

**Comment by Xiaomajohn** [comment]
yes,i use vite for the extendsion,
for this simple test,i use a  vue file,I defined a message-sending function and bound it to a button. After clicking it, a message is sent to the background of the extension plugin.
`const debugUrl = () => {
  chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
    console.log('tabs', tabs);
    chrome.runtime.sendMessage({ type: 'debug_url', url: tabs[0] });
  });
}
</script>

<template>
  <main>
    <h3>Popup Page</h3>

    <div class="calc">
      <li>
        <button @click="debugUrl">debug_url</button>

      </li>
      <button @click="minus" :disabled="count <= 0">-</button>
      <label>{{ count }}</label>
      <button @click="add">+</button>
    </div>

    <a :href="link" target="_blank"> generated by create-chrome-ext </a>
  </main>
</template>`

And in the background,I'm using a listener that listens to the specified key：debug-url,But when I click on it, the console only prints to the first position



`import { crx } from 'playwright-crx';
import { createTodos } from './todos.ts';

chrome.runtime.onMessage.addListener(async (request) => {
  if (request.type === 'debug_url') {
    chrome.runtime.sendMessage({ type: 'debug_url', url: request.tab?.url });
    await chrome.action.disable();
    console.log('first');
    const crxApp = await crx.start({ slowMo: 500 });//----err position is there
    console.log('second');

    const page = await crxApp.attach(request.tab?.id!).catch(() => crxApp.newPage());
    console.log('3');

    try {
      await createTodos(page);
    } finally {
      await crxApp.close();
      await chrome.action.enable();
    }
  }
});`

**Comment by ruifigueira** [comment]
Do you have the `debugger` permission on your extension manifest?

It should look like this:

https://github.com/ruifigueira/playwright-crx/blob/main/examples/todomvc-crx/public/manifest.json#L12

**Comment by Xiaomajohn** [comment]
Yes, I left out the debugger, but I was wondering if the library had documentation to guide me. I found out that crxApp can't create objects for the current page, only create new ones
When I just wanted to locate the current page and operate on the current page based on the tabId passed by the extension, I found that there was no such api to implement it

`    const crxApp = await crx.start({ slowMo: 500 });
    const page = await crxApp.attach(request.tab?.id!).catch(() => crxApp.newPage());
`
![image](https://github.com/user-attachments/assets/9af5b654-287a-4110-af4e-e9f0d3b2e7ea)

**Comment by Xiaomajohn** [comment]
Is that I forgot to add permissions to the extension, after adding it, it is OK
Thank you for your help！！

**Comment by Xiaomajohn** [comment]
Thank you for your help！！
Nice project‌
